c++/src/kj/string.h

```diff
@@ -278,14 +278,18 @@ class String {
   inline bool operator!=(decltype(nullptr)) const { return content.size() > 1; }
 
   inline bool operator==(const StringPtr& other) const { return StringPtr(*this) == other; }
+#if !__cpp_impl_three_way_comparison
   inline bool operator!=(const StringPtr& other) const { return StringPtr(*this) != other; }
+#endif
   inline bool operator< (const StringPtr& other) const { return StringPtr(*this) <  other; }
```

Comment:
strictly speaking all these comparison have to be part of the previous ifdef?

Reply:
Nope, we still need them -- the compiler will only synthesize "ordering" comparison operators if we define a spaceship operator -- which I think we should do, but figured I'd do the conservative thing for now.

I found this article helpful when learning about the new rules, by the way: https://brevzin.github.io/c++/2019/07/28/comparisons-cpp20/